plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/UserdataMapper.java

```diff
@@ -287,23 +295,49 @@ public static boolean hasSerializedModule(IEObjectDescription eObjectDescription
 	 */
 	public static void writeDependenciesToUserData(N4JSResource resource, Map<String, String> userData) {
 		final Set<URI> dependencies = Sets.newLinkedHashSet();
-		computeCrossRefs(resource, targetObj -> {
-			final Resource targetRes = targetObj.eResource();
-			if (targetRes != null) {
-				dependencies.add(targetRes.getURI());
-			}
-		});
-		userData.put(USERDATA_KEY_DEPENDENCIES, joiner.join(dependencies));
+		final Set<URI> dependenciesLoadTimeForInheritance = Sets.newLinkedHashSet();
+		computeCrossRefs(resource, dependencies::add);
+		getDependenciesLoadTimeForInheritance(resource, dependenciesLoadTimeForInheritance::add);
```

Comment:
This difference in naming was intentional. As you expected, the old `compute...` method is traversing the entire AST and computing the value it returns; the new `get...` method simply returns some information that is already in the TModule; it just modifies the values slightly (filter + map module to resource URI). It just looks like a lot because 1) Java is so verbose and 2) there is some error checking.

Reply:
If it is a getter, why don't you let it return something? Is there a reason to use an acceptor here? If so, why call it a getter?

Reply:
I used the acceptor for consistency with the existing method. Removing acceptor argument from both methods (old and new).